rest/context.go

```diff
@@ -143,10 +142,8 @@ type RequestContext interface {
 	ResponseWriter() http.ResponseWriter
 }
 
-// gorillaRequestContext is an implementation of the RequestContext interface. It wraps
-// Gorilla's context package from which it attempts to store/retrieve values, delegating
-// to the parent Context if necessary.
-type gorillaRequestContext struct {
+// requestContext is an implementation of the RequestContext interface.
+type requestContext struct {
 	context.Context
```

Comment:
maybe it's better not to embed the interface here and enforce that req.Context() is used

Reply:
What I mean is that I think only a single context should be used. Rather than having two contexts, the embedded one and the http.Request's context. This might lead to more code changes and breaking changes though 🤷